[InputStream] Add support to audio codec profiles /  ClearKey crypto type

CastagnaIT opened the pull request:
## Description
<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here. -->
In reference to opened PR #23524 this allow InputStream binary addons to add the codec profile to audio streams
mainly to display (by using PR #23524) the appropriate codec name in the audio track list

I have already predisposed interface with the new codec profiles that will be available with the next ffmpeg version
    DTSCodecProfileHDMAX
    DTSCodecProfileHDMAIMAX
    DDPlusCodecProfileAtmos
so when we will switch to it there is no need to rebuild all the addons but only set the appropriate profile conversion on kodi core

---

This PR include commits of #23540 to add ClearKey as a valid type for encryption scheme like existing Widevine, PlayReady and WisePlay.


## How has this been tested?
<!--- Please describe in detail how you tested your change. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Builded locally just to test that the value is set, ISAdaptive implementation will be done after merged
## What is the effect on users?
<!--- Summarize the effect of this change on Kodi end-users. -->
<!--- If the PR does not have a noticeable impact (e.g., if it only changes documentation), -->
<!--- just leave it empty. Put in more detail the bigger the impact is. -->
<!--- This section may be used for automatic creation of release notes. -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` with no space in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

phunkyfish commented:
This conflicts with https://github.com/xbmc/xbmc/pull/23540 in that the version bumps conflict.

Can the changes be done in a single PR with 3 commits, the last being the version bump?

phunkyfish commented:
The change itself looks to be ok. But it is customary to have at least one addon use the API change lined up with a PR before merging the Kodi.

CastagnaIT commented:
I didn't prepare it yet because there are too many PRs opened on ISAdaptive that touch different parts of it, and no one that can review PR's, PR stand opened for more than a week (yesterday by luck glenn guy review it, but i had not time)
also the other "clearkey" PR its not implemented on ISAdaptive

fwir features cannot be merged on Beta so we need these on this last kodi Alpha

in next days or next week i will do the ISA PR for audio codecs profiles,
for the "clearkey" implementation on ISA it is possible that glennguy wants to wait after the big decrypter refactor PR be merged
It will take some time because it is a huge PR

> Can the changes be done in a single PR with 3 commits, the last being the version bump?

i can merge the "clearkey" PR with this one soon

CastagnaIT committed:
[addons] increase global API to 3.3.0, videocodec to 2.1.0

phunkyfish commented:
Ok, I understand. Let’s get the 2 PRs into one so there is a single verison change and we can go from there.

As this is solely for inputstreams I think we can forgo normal protocols and make an exception.

phunkyfish reviewed (approved):
Please clean up the clang format suggestions. Otherwise this is good.

glennguy committed:
[Android] support ClearKey crypto type

CastagnaIT commented:
just to let you know that i opened the PR on ISAdaptive to implement audio profiles and tested
clang has been applied

phunkyfish commented:
Ok, I think we should go ahead and merge this. We can fix the Apple errors on ffmpegdirect afterwards.

CastagnaIT commented:
ouch i forgot a change in the CDVDDemuxClient for profiles i need some time

CastagnaIT committed:
[DVDDemuxClient] Allow update profile on stream reset

When InputStream interface do OpenStream addon callback the addon may change codec but also profiles,
this is the case with InputStream Adaptive since it initially set stream info by using the manifest data, but the demuxer that act after may change these info (so reset stream properties)

CastagnaIT commented:
done

AlwinEsch commented on xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon-instance/inputstream/stream_codec.h:
```diff
@@ -141,6 +141,28 @@ extern "C"
+
+    // Audio codec profiles
+
+    AACCodecProfileMAIN,
```
Only a bit about API descriptions missing on new added values :smirk:

CastagnaIT replied:
description added for each one

AlwinEsch commented on xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon-instance/inputstream/stream_codec.h:
```diff
@@ -141,6 +141,28 @@ extern "C"
+    MPEG2AACCodecProfileHE,
+    DTSCodecProfile,
+    DTSCodecProfileES,
+    DTSCodecProfile9624,
```
Specially here, as firstly I become a big thinking in mind whats "9624" means :sweat_smile::thinking: and comes then the "96 kHz / 24 Bit" in mind.

AlwinEsch reviewed (approved):
Mainly OK, only about docs, but can also leaved as is.

CastagnaIT committed:
[InputStream] Add support to audio codec profiles
